packages/console/src/utilities/validator.ts

```diff
@@ -1,12 +1,16 @@
-export const uriValidator = (value: string) => {
+import { ApplicationType } from '@logto/schemas';
+import { redirectNativeRegEx, redirectUriRegEx } from '@logto/shared';
+
+export const uriValidator = (value: string, applicationType?: ApplicationType) => {
```

Comment:
How about passing in a regular expression with the second param, instead of the application type? Because logically, a `uriValidator` has nothing to do with application type.
We can make the regex param optional and use `uriRegEx` as default value. E.g.:
```ts
export const uriValidator = (value: string, regex = uriRegEx) => {
  ...
}
```

Reply:
we can do but it just increases more lines of code. we need to import the regex and do the conditional pass there like that. Here it's just an additional parameter pass. still we can do that but I prefer this way, inputs?

Reply:
I prefer the way I suggested, as we can keep the logic pure and de-coupled, that a url validator should not have anything to do with application type. Please revise the code accordingly if that makes sense to you.